src/github.com/getlantern/eventual/eventual.go

```diff
@@ -52,15 +57,20 @@ func (v *value) Set(val interface{}) {
 func (v *value) processUpdates() {
 	for val := range v.updates {
 		v.val.Store(val)
-		if !v.gotFirst {
+		if v.gotFirst == FALSE {
 			// Signal to blocking callers that we have the first value
 			v.wg.Done()
-			v.gotFirst = true
+			v.gotFirst = TRUE
 		}
 	}
 }
 
 func (v *value) Get(timeout time.Duration) (interface{}, bool) {
+	if atomic.LoadInt32(&v.gotFirst) == TRUE {
+		// Short-cut used once value has been set, to avoid extra goroutine
+		return v.val.Load(), true
+	}
```

Comment:
@myleshorton This optimization speeds up Get() by 2 orders of magnitude.  I guess spawning goroutines does cost something.  Time for op went from about 2 microseconds to 20 nanoseconds.